File: mirea/cources/2026-2027/aii-july/01/Part-04-Binary-search/examples/example-12.py
# Первый элемент, не меньший x, в Python3

# Чтение отсортированного массива и значения x
n = int(input())
a = list(map(int, input().split()))
x = int(input())

# Ищем минимальный индекс, на котором a[index] >= x
left = 0
right = n
while left < right:
    mid = (left + right) // 2
    if a[mid] < x:
        left = mid + 1
    else:
        right = mid

# Выводим позицию и значение найденного элемента или -1
if left == n:
    print(-1)
else:
    print(left, a[left])

File: mirea/cources/2026-2027/aii-july/01/Part-04-Binary-search/examples/example-12.cpp
// Первый элемент, не меньший x, в C++

#include <iostream>
#include <vector>

using namespace std;

int main() {
    // Чтение отсортированного массива и значения x
    int n;
    cin >> n;

    vector<int> a(n);
    for (int i = 0; i < n; i++) {
        cin >> a[i];
    }

    int x;
    cin >> x;

    // Ищем минимальный индекс, на котором a[index] >= x
    int left = 0;
    int right = n;
    while (left < right) {
        int mid = left + (right - left) / 2;
        if (a[mid] < x) {
            left = mid + 1;
        } else {
            right = mid;
        }
    }

    // Выводим позицию и значение найденного элемента или -1
    if (left == n) {
        cout << -1 << '\n';
    } else {
        cout << left << ' ' << a[left] << '\n';
    }
}